api/auth/auth-middleware.js

```diff
@@ -63,7 +101,28 @@ const validateRoleName = (req, res, next) => {
       "message": "Role name can not be longer than 32 chars"
     }
   */
-  next();
+const validateRoleName = async (req, res, next) => {
+  try {
+    const { role_name } = req.body;
+    if (role_name) {
+      const trimmedRoleName = role_name.trim();
+      req.roleName = trimmedRoleName;
+
+      if (trimmedRoleName === "admin") {
+        return res.status(422).json({ message: "Role name can not be admin." });
+      }
+      if (trimmedRoleName.length > 32) {
+        return res
+          .status(422)
+          .json({ message: "Role name can not be longer than 32 chars." });
+      }
```

Comment:
imi pare ca lipseste un next() aici sau ma gresesc?

Reply:
dupa ce ai facut toate verificarile unui `role_name` vrei sa transmiti utilizatorul mai departe

Reply:
Inteleg la ce va referiti. Initial era avem un next(), dar problema era ca chiar daca nimeream intr-un if cu restrictii, oricum se creea utilizatorul. Si de aceea am scos acel next. Dar probabil nu l-am pus eu unde trebuie, presupun ca in interiorul primului if trebuie. O sa verific asta.